mlrun/runtimes/sparkjob/abstract.py

```diff
@@ -603,9 +603,14 @@ def _update_igz_jars(self, deps):
                 self.spec.deps["files"] = []
             self.spec.deps["files"] += deps["files"]
 
-    def with_igz_spark(self):
+    def with_igz_spark(self, mount_v3io_to_executor=True):
         self._update_igz_jars(deps=self._get_igz_deps())
         self.apply(mount_v3io_extended())
+        self.spec.driver_volume_mounts = self.spec.volume_mounts
```

Comment:
I think we should account for the cases where there were mounts prior to calling this - this may certainly happen in several cases and we should be able to handle it properly. 
This would mean:
1. Keeping the existing `self.spec.volume_mounts` in a temp variable, and clearing it prior to calling the `apply()`.
2. After copying the results to driver/executor mounts - copy into `spec.volume_mounts` back from the temp variable (rather than putting `[]` in it).

I'm also concerned about the case where someone already put something in `driver_volume_mounts` or `executor_volume_mounts` - they are updated also when calling `with_executor_host_path_volume` and possibly elsewhere.
to handle these cases need to append the mounts from the `spec.volume_mounts` rather than just copy them.